### PR TITLE
[PT FE]: fix aten::sum for uint8 input

### PR DESCRIPTION
**Pull request by eaidova**
### Details:
 - *fix accuracy of aten::sum translation for uint8 inputs*
 - *update aten::sum support with type and out arguments*

### Tickets:
 - *CVS-123277*

### CODE REVIEW

File: tests/layer_tests/pytorch_tests/test_sum.py [hunk 1]
```diff
@@ -7,72 +7,97 @@
...
-    def create_model(self, axes, keep_dims):
+        import numpy as np
+        min_value = -10 if input_dtype not in ["uint8", "bool"] else 0
+        max_value = 10 if input_dtype != "bool" else 2
+        input = np.random.randint(min_value, max_value, (1, 3, 5, 5)).astype(input_dtype)
+        if not out:
+            return (input, )
+        if out_dtype is None:
+            out_dtype = input_dtype if input_dtype not in ["uint8", "bool"] else "int64"
+        out = np.zeros((1, 3, 224, 224), dtype=out_dtype)
```
Comment (eaidova) [inline tests/layer_tests/pytorch_tests/test_sum.py:20]:
```suggestion
        out = np.zeros((1, 3, 5, 5), dtype=out_dtype)
```

File: src/frontends/pytorch/src/op/sum.cpp [hunk 2]
```diff
@@ -13,25 +13,60 @@ namespace pytorch {
 namespace op {
 
 OutputVector translate_sum(const NodeContext& context) {
-    num_inputs_check(context, 1, 3);
+    // aten::sum.dim_IntList(Tensor self, int[1]? dim, bool keepdim=False, *, ScalarType? dtype=None) -> Tensor
+    // aten::sum(Tensor self, *, ScalarType? dtype=None) -> Tensor
+    // aten::sum.IntList_out(Tensor self, int[1]? dim, bool keepdim=False, *, ScalarType? dtype=None, Tensor(a!) out) ->
+    // Tensor(a!) aten::sum.out(Tensor self, *, ScalarType? dtype=None, Tensor(a!) out) -> Tensor(a!)
```
Comment (eaidova) [inline src/frontends/pytorch/src/op/sum.cpp:19]:
```suggestion
    // Tensor(a!) 
    // aten::sum.out(Tensor self, *, ScalarType? dtype=None, Tensor(a!) out) -> Tensor(a!)
```

File: tests/layer_tests/pytorch_tests/test_sum.py [hunk 3]
```diff
@@ -7,72 +7,97 @@
...
+                    return torch.sum(x, self.axes, self.keep_dims, dtype=self.dtype, out=out), out
+                else:
+                    return torch.sum(x, self.axes, self.keep_dims, out=out), out
 
         ref_net = None
 
-        return aten_sum(axes, keep_dims), ref_net, "aten::sum"
+        return aten_sum(input_torch_dtype, axes, keep_dims, torch_dtype, out), ref_net, "aten::sum"
 
     @pytest.mark.parametrize("axes,keep_dim",
```
Comment (PiotrKrzem) [inline tests/layer_tests/pytorch_tests/test_sum.py:92]: Overall, I see no issues :) My suggestion is to to rename:
```suggestion
    @pytest.mark.parametrize("axes,keep_dims",
```
To keep consistency with names used in the rest of the code.

**Review by PiotrKrzem — APPROVED** [review]
